File: clients/GVGAI-PythonClient/src/IOSocket.py
import logging
import os
import socket
import sys
import traceback

import time


class IOSocket:
    """
     * Socket for communication
    """

    def __init__(self, port):
        self.BUFF_SIZE = 1024000
        self.TOKEN_SEP = '#'
        self.port = port
        self.hostname = "127.0.0.1"
        self.logfilename = "./logs/clientLog.txt"
        self.connected = False
        self.socket = None
        self.logfile = open(self.logfilename, "a")

    def initBuffers(self):
        while not self.connected:
            try:
                self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
                print ("Selected host is: " + str(self.hostname))
                print ("Selected port is: " + str(self.port))
                self.socket.connect((self.hostname, self.port))
                self.connected = True
                print ("Client connected to server [OK]")
            except Exception as e:
                logging.exception(e)
                print("Client connected to server [FAILED]")
                traceback.print_exc()
                sys.exit()

    def writeToFile(self, line):
        sys.stdout.write(line + os.linesep)
        self.logfile.write(line + os.linesep)
        sys.stdout.flush()
        self.logfile.flush()

    def writeToServer(self, messageId, line, log):
        msg = str(messageId) + self.TOKEN_SEP + line + "\n"
        print(msg)

        try:
            self.socket.send(bytes(msg))
            if log:
                self.writeToFile(msg)
        except Exception as e:
            logging.exception(e)
            print ("Write " + self.logfilename + " to server [FAILED]")
            traceback.print_exc()
            sys.exit()

    def readLine(self):
        try:
            # return self.recv_timeout
            return self.socket.recv(1024000)
        except Exception as e:
            logging.exception(e)
            print ("Read from server [FAILED]")
            traceback.print_exc()
            sys.exit()

File: clients/GVGAI-JavaClient/src/utils/IOSocket.java
package utils;

import java.io.*;
import java.net.ConnectException;
import java.net.Socket;
import java.util.Scanner;

/**
 * Created by dperez on 23/05/2017.
 */
public class IOSocket extends IO {


    private Socket socket;
    private Scanner in;
    private PrintStream out;

    private String hostname = "localhost";
    private int port;

    public IOSocket(int port)
    {
        super();
        this.port = port;
    }

    /**
     * Creates the buffers for pipe communication.
     */
    @Override
    public void initBuffers() {

        boolean connected = false;
        try {

            while(!connected)
            {
                try{
                    socket = new Socket(hostname, port);
                    connected = true;
                    System.out.println("Client connected to server [OK]");
                }catch (ConnectException e) {
                    //System.out.println(e);
                }
            }

            out = new PrintStream(socket.getOutputStream());
            in = new Scanner(socket.getInputStream());

        } catch (Exception e) {
            System.out.println("Exception creating the client process: " + e);
            e.printStackTrace();
        }

    }


    /**
     * Writes a line to the server, adding a line separator at the end.
     * @param messageId the server is expecting.
     * @param line to write
     * @param log if true, write to file as well.
     */
    @Override
    public void writeToServer(long messageId, String line, boolean log)
    {
        String msg = messageId + ClientComm.TOKEN_SEP + line;
        this.writeToServer(msg);
        if(log) this.writeToFile(msg);
    }

    @Override
    public String readLine() throws IOException{
        return in.nextLine();
    }

    /**
     * Writes a line to the server, adding a line separator at the end.
     * @param line to write
     */
    @Override
    protected void writeToServer(String line)
    {
        try {
            out.print(line + lineSep);
            out.flush();
        }catch(Exception e)
        {
            System.out.println("Error trying to write " + line + " to the server.");
            e.printStackTrace();
        }
    }




}
